app/controllers/welcome_controller.rb

```diff
@@ -20,13 +20,16 @@ def goodbye
   end
 
   def user_home
-    bikes = current_user.bikes
-    page = params[:page] || 1
-    @locks_active_tab = params[:active_tab] == 'locks'
-    @per_page = params[:per_page] || 20
-    paginated_bikes = Kaminari.paginate_array(bikes).page(page).per(@per_page)
-    @bikes = BikeDecorator.decorate_collection(paginated_bikes)
-    @locks = LockDecorator.decorate_collection(current_user.locks)
+
+    unless return_to_if_present
```

Comment:
This shouldn't be necessary, we're already [authenticating the user](https://github.com/bikeindex/bike_index/blob/master/app/controllers/welcome_controller.rb#L3) (which [redirects them to accept the terms](https://github.com/bikeindex/bike_index/blob/master/app/controllers/concerns/authentication_helper.rb#L6) if they haven't).